Sources/PubNub/Networking/HTTPRouter.swift

```diff
@@ -32,6 +32,10 @@ public protocol RouterConfiguration {
   var useRequestId: Bool { get }
   /// Ordered list of key-value pairs which identify various consumers.
   var consumerIdentifiers: [String: String] { get }
+  /// This controls whether to enable a new, experimental implementation of Subscription and Presence handling
+  var enableEventEngine: Bool { get }
+  /// When `true` the SDK will resend the last channel state that was set using `PubNub.setPresence`
+  var maintainPresenceState: Bool { get }
```

Comment:
You're absolutely right. The problem in Swift SDK is that there are additional protocols like `RouterConfiguration` or even `SubscriptionConfiguration` that allow you to override default `PubNubConfiguration` by the user. In fact, the associated configuration for any class that inherits from `HTTPRouter` is `RouterConfiguration`, not `PubNubConfiguration`. I already deprecated `SubscriptionConfiguration`, and it looks like removing `RouterConfiguration` could be achieved while working on modernization. Although I also don't like it, I couldn't remove it right now since it requires huge changes. Is it OK for you to address this issue for the modernization epic and align it with all other SDKs?

Reply:
I believe that PubNub Swift SDK is the only SDKs which allows customized practically everything. But yeah, we can think on it later.